R/modified_update.R

```diff
@@ -307,7 +307,7 @@ mu_build_data <- function(con, index, meta, pop) {
   if (is.na(x$coverage_set_new)) {
     stop("Import error: no new coverage found")
   } else {
-    if(meta$touchstone_mod$touchstone_name == meta$touchstone_use)  {
+    if(meta$touchstone_mod$touchstone_name == meta$group$touchstone_name[meta$group$index == index])  {
```

Comment:
Surely this breaks previous reports that used the modified update function?  Can you please check?  We'll get some automated tests in at some point but that's going to be a bit of a pain unfortunately

Reply:
This bit of code was originally added for report interenal-2017-modup2-201510gavi-201510gavi, for which touchstone_mod = touchstone_use. It has no impact on other reports that use jenner (lucky). But this causes warning messages for report modup-201710 where there are three touchstone_use leading to 'only the first one is used' type of warning message.

Reply:
That will need eliminating - can you point me at the report that triggers if it it's not obvious how to make it do the right thing?

Reply:
The warning is eliminated by the change I made last night.

Basically, in the report below (which triggers warning)
https://github.com/vimc/montagu-reports/blob/master/src/modup-201710/modified_update.R
```
touchstone_name_mod <- "201710gavi"
touchstone_use <- c("201510gavi",
                    "201310gavi",
                    "201210gavi-201303gavi")
```
so the logical operator '==' triggers the warning. However, since R only goes through the 'if' clause and then jump to 'else' where the right chunk of code is ran, it is doing the right think luckily.

I have changed 
```
meta$touchstone_mod$touchstone_name == meta$touchstone_use 
```
to 
```
meta$touchstone_mod$touchstone_name == meta$group$touchstone_name[meta$group$index == index]
```
where index in the function is of length 1. The problem is solved.

Reply:
I did 
```
report <- "modup-201710"
orderly::orderly_run(report)
```
This produces waning messages.

Then
```
devtools::load_all("../jenner") # current version in this PR
report <- "modup-201710"
orderly::orderly_run(report)
```
This eliminates warning messages, and produces the same impact numbers as the previous one.

Reply:
Thanks - well caught